src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSink.java

```diff
@@ -248,9 +308,14 @@ private static void mergeHeaderlessBamShards(String outputFile, SAMFileHeader he
         fs.delete(outputPath, true);
 
         try (final OutputStream out = fs.create(outputPath)) {
-            new SAMOutputPreparer().prepareForRecords(out, SAMFormat.BAM, header); // write the header
+            new SAMOutputPreparer().prepareForRecords(out, samOutputFormat, header); // write the header
```

Comment:
mergeHeaderlessBamShards is only called from writeReadsSingle, which passes the same header through to both the output class and the merging code. This reminds me, though, that we should change Hadoop-BAM (as well as GATK which we've previously discussed) to use setHeaderStrict.